lib/persist-state.c

```diff
@@ -267,6 +301,13 @@ _alloc_value(PersistState *self, guint32 orig_size, gboolean in_use, guint8 vers
   persist_state_unmap_entry(self, self->current_ofs);
 
   self->current_ofs += size + sizeof(PersistValueHeader);
+
+  if (!_check_watermark(self) && !_grow_store(self, self->current_size + PERSIST_FILE_INITIAL_SIZE))
+    {
+      msg_error("Can't preallocate space for persist file, syslog-ng is stopping", NULL);
+      main_loop_exit();
```

Comment:
My idea is, that I will extend the apphook with an apphook_request_shutdown and its state AH_REQUEST_SHUTDOWN. The main loop can subscribe for this.
@bazsi, what is your opinion about this idea?